desc/equilibrium/coords.py

```diff
@@ -376,131 +382,147 @@ def fixup(x, *args):
+        Shape (lmbda.grid.num_rho, )
+
+    Returns
+    -------
+    lmbda_minus_iota_omega, modes
+        Spectral coefficients and modes.
+        Shape (num rho, num zeta, num modes).
+
+    """
+    grid = lmbda.grid
+    errorif(not grid.fft_poloidal, NotImplementedError, msg="See note in docstring.")
+    # TODO(#1243): assert grid.sym==eq.sym once basis is padded for partial sum
+    # TODO: (#568)
+    warnif(
+        grid.M > lmbda.basis.M,
+        ResolutionWarning,
+        msg="Poloidal grid resolution is higher than necessary for coordinate mapping.",
+    )
+    warnif(
+        grid.M < lmbda.basis.M,
+        ResolutionWarning,
+        msg="High frequency lambda modes will be truncated in coordinate mapping.",
+    )
+    lmbda_minus_iota_omega = lmbda.transform(L_lmn)
+    lmbda_minus_iota_omega = (
+        rfft(grid.meshgrid_reshape(lmbda_minus_iota_omega, "rzt"), norm="forward")
+        .at[..., (0, -1) if ((grid.num_theta % 2) == 0) else 0]
+        .divide(2)
+        * 2
+    )
+    return lmbda_minus_iota_omega, jnp.fft.rfftfreq(grid.num_theta, 1 / grid.num_theta)
+
+
+@partial(jit, static_argnames=["tol", "maxiter"])
 def _map_clebsch_coordinates(
-    coords,
     iota,
+    alpha,
+    zeta,
     L_lmn,
-    L_basis,
-    guess=None,
-    period=np.inf,
+    lmbda,
+    theta0=None,
+    *,
     tol=1e-6,
     maxiter=30,
-    full_output=False,
     **kwargs,
 ):
     """Find θ for given Clebsch field line poloidal label α.
 
     Parameters
     ----------
-    coords : ndarray
-        Shape (k, 3).
-        Clebsch field line coordinates [ρ, α, ζ]. Assumes ζ = ϕ.
-        Each row is a different point in space.
     iota : ndarray
-        Shape (k, ).
+        Shape (num iota, ).
         Rotational transform on each node.
+    alpha : ndarray
+        Shape (num alpha, )
+    zeta : ndarray
+        Shape (num zeta, )
     L_lmn : jnp.ndarray
-        Spectral coefficients for lambda.
-    L_basis : Basis
-        Spectral basis for lambda.
-    guess : jnp.ndarray
-        Shape (k, ).
+        Spectral coefficients for λ.
+    L : Transform
+        Transform for λ built on coordinates [ρ, θ, ζ].
+    theta0 : jnp.ndarray
+        Shape (num iota, num alpha, num zeta).
         Optional initial guess for the computational coordinates.
-    period : float
-        Assumed periodicity for α.
-        Use ``np.inf`` to denote no periodicity.
     tol : float
         Stopping tolerance.
     maxiter : int
         Maximum number of Newton iterations.
-    full_output : bool, optional
-        If True, also return a tuple where the first element is the residual from
-        the root finding and the second is the number of iterations.
     kwargs : dict, optional
         Additional keyword arguments to pass to ``root_scalar`` such as ``maxiter_ls``,
         ``alpha``.
 
     Returns
     -------
-    out : ndarray
-        Shape (k, 3).
-        DESC computational coordinates [ρ, θ, ζ].
-    info : tuple
```

Comment:
Why did you remove info? This is not a good coding practice. This is basically passive encryption.

Reply:
What happens if root finding fails for some reason and I want to debug it.

Reply:
If the residual is calculated elsewhere, ignore this comment.

Reply:
This is a new function, see https://github.com/PlasmaControl/DESC/pull/1826#discussion_r2275331172, so I disagree with the statement about coding practice. Whatever code users have that uses root finding, nothing has changed. They can still get their `info` tuple.

In this new function, I have not added functionality to return auxiliary information about the root finding simple because it is impossible - functions that decorated with `jnp.vectorize` must return arrays.

Reply:
I am not convinced that this root finding can fail by the way.